cdap-runtime-ext-remote-hadoop/src/main/java/io/cdap/cdap/runtime/spi/provisioner/remote/JavaSocketEdgeNodeCheck.java

```diff
@@ -0,0 +1,41 @@
+package io.cdap.cdap.runtime.spi.provisioner.remote;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.util.List;
+
+public class JavaSocketEdgeNodeCheck {
+  private static final Logger LOG = LoggerFactory.getLogger(JavaSocketEdgeNodeCheck.class);
+  private static final int CHECK_PORT = 22;
```

Comment:
can we take port number from the hostname, if it is set? 


like ->  host1:23, host6:55 ?

Reply:
We can, but it is not the only place in the code where port should be set for this option. Also appfabric and runtime-spi jar should be changed for that. It makes sense to do these changes in the scope of a separated feature.